subsys/net/lib/shell/cm.c

```diff
@@ -0,0 +1,958 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(net_shell);
+
+#include <zephyr/net/net_if.h>
+#include <strings.h>
+#include <ctype.h>
+
+#include "net_shell_private.h"
+#include "conn_mgr_private.h"
+#include <zephyr/net/conn_mgr_connectivity.h>
+#include <zephyr/net/conn_mgr_monitor.h>
+
+
+#define CM_IF_NAME_NONE "unnamed"
+
+#if defined(CONFIG_NET_INTERFACE_NAME)
+#define CM_MAX_IF_NAME MAX(sizeof(CM_IF_NAME_NONE), CONFIG_NET_INTERFACE_NAME_LEN)
+#else
+#define CM_MAX_IF_NAME sizeof(CM_IF_NAME_NONE)
+#endif
+
+#define CM_MAX_IF_INFO (CM_MAX_IF_NAME + 40)
+
+/* Parsing and printing helpers */
+
+enum cm_type {
+	CM_TARG_IFACE,
+	CM_TARG_NONE,
+	CM_TARG_ALL,
+	CM_TARG_INVALID,
+};
+struct cm_target {
+	enum cm_type type;
+	struct net_if *iface;
+};
+enum cm_gs_type {
+	CM_GS_GET,
+	CM_GS_SET
+};
+struct cm_flag_string {
+	char *name;
+	enum conn_mgr_if_flag flag;
+};
+
+/* The first entry in each flag alias group will be used for human-readable shell
+ * output. The rest are accepted synonyms.
+ */
+static const struct cm_flag_string flag_strings[] = {
+	{"PERSISTENT",		CONN_MGR_IF_PERSISTENT},
+	{"persist",		CONN_MGR_IF_PERSISTENT},
+	{"p",			CONN_MGR_IF_PERSISTENT},
```

Comment:
Do we need this fancy short name handling? The shell has support for dynamic shell completion so user does not need to type full string.

Reply:
It is probably not needed. I will look into what would be required for dynamic shell completion. But I would guess that would be similarly complex to implement, since Zephyr needs to be made aware of the possible auto-completions

Reply:
To take advantage of dynamic command completion would require a significant or complete restructuring of this command set, which I am not eager to undertake.

For the time being, I will simply remove the flag name aliases